packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/template.html

```diff
@@ -63,9 +63,82 @@
+        data-cancel-text="{{ ::'octavia_load_balancer_pools_detail_members_add_ip_instance_cancel_action' | translate }}"
+        data-loading="$ctrl.isLoading"
+        data-prevent-next="true"
+        data-on-focus="$ctrl.onStep2Focus()"
+        data-on-cancel="$ctrl.goBack()"
     >
+        <div
+            class="oui-box box-member mb-4"
+            data-ng-repeat="member in $ctrl.model.members track by $index"
+            ng-init="memberIndex = $index"
+        >
+            <div class="row align-items-center">
+                <oui-field
+                    class="col-lg-5 col-md-5 col-sm-12"
+                    data-label="{{:: 'octavia_load_balancer_pools_detail_members_add_ip_instance_name_label' | translate }}"
+                >
+                    <div class="oui-input-overlay">
+                        <input
+                            type="text"
+                            class="oui-input"
+                            name="{{:: 'member-name-' + memberIndex}}"
+                            data-ng-model="member.name"
+                        />
+                    </div>
+                </oui-field>
+                <oui-field
+                    class="col-lg-4 col-md-4 col-sm-12"
+                    data-label="{{:: 'octavia_load_balancer_pools_detail_members_add_ip_instance_ip_label' | translate }}"
+                >
+                    <div class="oui-input-overlay">
+                        <input
+                            type="text"
+                            class="oui-input"
+                            name="{{:: 'member-ip-' + memberIndex}}"
+                            data-ng-model="member.address"
+                            data-ng-disabled="true"
+                        />
+                    </div>
+                </oui-field>
+                <oui-field
+                    class="col-lg-3 col-md-3 col-sm-12"
+                    data-label="{{:: 'octavia_load_balancer_pools_detail_members_add_ip_instance_port_label' | translate }}"
+                    data-label-popover="{{:: 'octavia_load_balancer_pools_detail_members_protocol_port_tooltip' | translate }}"
+                    data-error-messages="{ existing: ('octavia_load_balancer_pools_detail_members_protocol_port_existing' | translate) }"
+                >
+                    <input
+                        class="oui-input"
+                        type="number"
+                        name="{{:: 'member-port-' + memberIndex}}"
+                        data-ng-model="member.protocolPort"
+                        data-ng-model-options="{ allowInvalid: true }"
+                        data-min="1"
+                        data-max="65535"
+                        data-ng-disabled="$ctrl.duplicatePort && memberIndex !== 0"
+                        data-ng-change="$ctrl.onPortChange(memberIndex)"
+                        data-ui-validate="{ existing: '!$ctrl.isIpAlreadyAssociatedWithPort(member.address, $value)' }"
```

Comment:
Why not using `valid` attributes from stepper component instead ? Adding a new library just for one field validation seems a bit heavy

Reply:
Using valid will disabled the "confirm" button and will not display a custom error.
I agree that adding a library (not that big btw) is not great, but to achieve a custom validation in this case the solution is pretty ugly: the only way to access the input to change its validity is by adding an encapsulating form to be able to access the one behind step-form and finaly access the input control.